File: OOP_Seminars/OOP_Sem7_20jan/pyton/Shapes.py
from abc import ABC, abstractmethod
from math import pi
from typing import List

class Shape(ABC):
    @abstractmethod
    def get_area():
        pass

class SquareShape(Shape):
    def __init__(self, side):
        self.side = side

    def get_area(self):
        return self.side * self.side

class CircleShape(Shape):
    def __init__(self, radius):
        self.radius = radius

    def get_area(self):
        return pi * self.radius * self.radius

shapes: List[Shape] = [SquareShape(1), CircleShape(1)]

for shape in shapes:
    print(shape.get_area())

File: OOP_Seminars/OOP_Sem7_20jan/javascript/Shapes.js

class Shape {
    getArea () { throw new Error ("You must implement getArea()"); }
}

class SquareShape extends Shape {
    constructor (side) {
        super ();
        this.side = side;
    }

    getArea () { return this.side * this.side; }
}

class CircleShape extends Shape {
    constructor (radius) {
        super ();
        this.radius = radius;
    }

    getArea () { return Math.PI * this.radius * this.radius; }
}

shapes = [new SquareShape (1), new CircleShape (1)]

for (const shape of shapes)
    console.log (shape.getArea ())
